code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -172,325 +421,364 @@
-			else to_chat(src, "<span class='notice'>Out of uses.</span>")
-	else to_chat(src, "<span class='notice'>That's not a machine.</span>")
 
+//Robotic Factory: Places a large machine that converts humans that go through it into cyborgs. Unlocking this ability removes shunting.
 /datum/AI_Module/large/place_cyborg_transformer
 	module_name = "Robotic Factory (Removes Shunting)"
 	mod_pick_name = "cyborgtransformer"
 	description = "Build a machine anywhere, using expensive nanomachines, that can convert a living human into a loyal cyborg slave when placed inside."
 	cost = 100
-	power_type = /mob/living/silicon/ai/proc/place_transformer
-	var/list/turfOverlays = list()
+	one_purchase = TRUE
+	power_type = /datum/action/innate/ai/place_transformer
+	unlock_text = "<span class='notice'>You make contact with Space Amazon and request a robotics factory for delivery.</span>"
+	unlock_sound = 'sound/machines/ping.ogg'
+
+/datum/action/innate/ai/place_transformer
+	name = "Place Robotics Factory"
+	desc = "Places a machine that converts humans into cyborgs. Conveyor belts included!"
+	button_icon_state = "robotic_factory"
+	uses = 1
+	auto_use_uses = FALSE //So we can attempt multiple times
+	var/list/turfOverlays
 
-/datum/AI_Module/large/place_cyborg_transformer/New()
-	for(var/i=0;i<3;i++)
-		var/image/I = image("icon"='icons/turf/overlays.dmi')
-		turfOverlays += I
+/datum/action/innate/ai/place_transformer/New()
 	..()
+	for(var/i in 1 to 3)
+		var/image/I = image("icon"='icons/turf/overlays.dmi')
+		LAZYADD(turfOverlays, I)
 
-/mob/living/silicon/ai/proc/place_transformer()
-	set name = "Place Robotic Factory"
-	set category = "Malfunction"
-	if(!canPlaceTransformer())
+/datum/action/innate/ai/place_transformer/Activate()
+	if(!owner_AI.can_place_transformer(src))
 		return
-	var/sure = alert(src, "Are you sure you want to place the machine here?", "Are you sure?", "Yes", "No")
-	if(sure == "Yes")
-		if(!canPlaceTransformer())
-			return
-		var/turf/T = get_turf(eyeobj)
-		var/obj/machinery/transformer/conveyor = new(T)
-		conveyor.masterAI = src
-		playsound(T, 'sound/effects/phasein.ogg', 100, 1)
-		var/datum/AI_Module/large/place_cyborg_transformer/PCT = locate() in current_modules
-		PCT.uses --
-		can_shunt = 0
-		to_chat(src, "<span class='warning'>You cannot shunt anymore.</span>")
-
-/mob/living/silicon/ai/proc/canPlaceTransformer()
-	if(!eyeobj || !isturf(src.loc) || !canUseTopic())
+	active = TRUE
+	if(alert(owner, "Are you sure you want to place the machine here?", "Are you sure?", "Yes", "No") == "No")
+		active = FALSE
+		return
+	if(!owner_AI.can_place_transformer(src))
 		return
-	var/datum/AI_Module/large/place_cyborg_transformer/PCT = locate() in current_modules
-	if(!PCT || PCT.uses < 1)
-		alert(src, "Out of uses.")
+	var/turf/T = get_turf(owner_AI.eyeobj)
+	var/obj/machinery/transformer/conveyor = new(T)
+	conveyor.masterAI = owner
+	playsound(T, 'sound/effects/phasein.ogg', 100, 1)
+	owner_AI.can_shunt = TRUE
```

Comment:
Stealth changes

Reply:
Probably an accident.

Reply:
Definitely an accident, and there's a fix PR up